src/n_fft.h

```diff
@@ -0,0 +1,261 @@
+/*
+    Copyright (C) 2024 Vincent Neiger
+
+    This file is part of FLINT.
+
+    FLINT is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.  See <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef N_FFT_H
+#define N_FFT_H
+
+#include "flint.h"
+#include "nmod.h"
+#include "nmod_vec.h"
+#include "ulong_extras.h"
```

Comment:
Looks like `n_mulmod_shoup` is the only dependency here, so I suppose this would work with
```suggestion
#include "ulong_extras.h"
```